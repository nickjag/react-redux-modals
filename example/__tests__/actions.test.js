import configureStore from 'redux-mock-store';
import { actions } from 'react-redux-modals';

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

describe('actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe('modalAgree', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(actions.showModal('MODAL_AGREE'));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('modalProps', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(actions.showModal('MODAL_PROPS', {
        onSubmit: (data) => data
      }));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});