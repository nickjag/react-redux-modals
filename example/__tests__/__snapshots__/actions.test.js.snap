// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions modalAgree Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "modalProps": undefined,
      "modalType": "MODAL_AGREE",
    },
    "type": "react-redux-modals/SHOW",
  },
]
`;

exports[`actions modalProps Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "modalProps": Object {
        "onSubmit": [Function],
      },
      "modalType": "MODAL_PROPS",
    },
    "type": "react-redux-modals/SHOW",
  },
]
`;
